### PR TITLE
Fix for ogre failing when material already exists

### PR DESCRIPTION
**Pull request by wolfv**
This came up in the RoboStack project where we're using a more recent OGRE

### CODE REVIEW

**Comment by wolfv** [comment]
For more context: https://medium.com/robostack/cross-platform-conda-packages-for-ros-fa1974fd1de3

And here we're working towards a full galactic build: https://github.com/robostack/ros-galactic

File: rviz_rendering/src/rviz_rendering/material_manager.cpp [hunk 1]
```diff
@@ -122,9 +122,9 @@ void MaterialManager::enableAlphaBlending(
 
 void MaterialManager::createDefaultMaterials()
 {
-  auto material = Ogre::MaterialManager::getSingleton().create(
+  auto retrieve_result = Ogre::MaterialManager::getSingleton().createOrRetrieve(
     "BaseWhiteNoLighting", "rviz_rendering");
-  material->setLightingEnabled(false);
+  std::dynamic_pointer_cast<Ogre::Material>(retrieve_result.first)->setLightingEnabled(false);
```
Comment (ahcorde) [inline rviz_rendering/src/rviz_rendering/material_manager.cpp]:
```suggestion
   auto material = std::dynamic_pointer_cast<Ogre::Material>(retrieve_result.first);
   if (material) {
     material->setLightingEnabled(false);
   }
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by wolfv** [comment]
I am fine with this change although it should generally never happen (I suppose) since the resource will either be created or retrieved.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14909/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9652/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12580/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15067/)

**Review by ahcorde — CHANGES_REQUESTED** [review]
uncrustify is failing https://build.ros2.org/job/Rpr__rviz__ubuntu_focal_amd64/174/

**Comment by ahcorde** [comment]
friendly ping @wolfv

**Comment by wolfv** [comment]
I was on vacation :) fixed now!

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15055/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9789/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12742/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15223/)

**Comment by ahcorde** [comment]
@wolfv do you need to backport this ?

**Pull request merged**